### PR TITLE
Better @command definition that can register tab-completion strings

### PR DESCRIPTION
**Pull request by Razesdark**
This PR moves the list of commands from a static list in `util.py` to an argument
that is passed to the @command function. This way we dont have a separate list of tab completion.

ref: #801

### CODE REVIEW

File: mps_youtube/commands/__init__.py [hunk 1]
```diff
@@ -11,10 +11,22 @@
 RS = r'(?:(?:repeat|shuffle|-[avfw])\s*)'
 PL = r'\S*?((?:RD|PL|LL)[-_0-9a-zA-Z]+)\s*'
 
-
-def command(regex):
+## @command decorator
+##
+## The @command decorator takes two arguments, one is a regex that
```
Comment (ids1024) [inline mps_youtube/commands/__init__.py]: It's not very important, but I wouldn't describe a polymorphic function as taking "two arguments"

File: mps_youtube/player.py [hunk 2]
```diff
@@ -73,6 +73,7 @@ def play(self, songlist, shuffle=False, repeat=False, override=False):
                                                          override=self.override,
                                                          softrepeat=self.softrepeat)
                 self._playsong()
+                self.song_no += 1
```
Comment (ids1024) [inline mps_youtube/player.py]: Uh, the changes in this file seem unrelated to the purpose of the commit or PR

**Comment by ids1024** [comment]
This may not the the ideal solution, but it's definitely an improvement.

It might be worth thinking about something that results in less redundancy. Not using regexes at all seems like one option, but I don't know how that could be avoided given all the things it's used for. But many of the commands follow the same format of \<command-name\> {\<argument\>}, so perhaps it could be simplified in some way.

File: mps_youtube/player.py (hunk 2 above)
Comment (vn-ki) [reply mps_youtube/player.py]: You should not do this. This is handled by `CmdPlayer.next`

Comment (Razesdark) [reply mps_youtube/player.py]: You're right, this is unrelated. It should not be here. I'll remove it.

File: mps_youtube/player.py [hunk 3]
```diff
@@ -89,10 +89,10 @@ def play(self, songlist, shuffle=False, repeat=False, override=False):
             if config.SET_TITLE.get:
                 util.set_window_title("mpsyt")
 
+            print("songlistlen " + str(self.song_no) + ":" + str(len(songlist)))
```
Comment (ids1024) [inline mps_youtube/player.py]: Still have this line (which presumably is debugging code not meant to be committed) and another change in this file.

File: mps_youtube/commands/__init__.py (hunk 1 above)
Comment (Razesdark) [reply mps_youtube/commands/__init__.py]: Yeah, that was kind of embarrassingly bad.
I've updated it.
